src/features/foldingProvider.ts

```diff
@@ -48,6 +51,34 @@ export default class AsciidocFoldingRangeProvider implements vscode.FoldingRange
     return listOfRanges
   }
 
+  private static getOpenBlockFoldingRanges (document: vscode.TextDocument) {
+    const listOfRanges = []
+    const openBlockIndexes = []
+    let end: number | undefined
+    for (let lineIndex = 0; lineIndex < document.lineCount; lineIndex++) {
+      const line = document.lineAt(lineIndex)
+      if (line.text === '--') {
+        openBlockIndexes.push(lineIndex)
+      }
+      if (line.text === '[source]') {
+        openBlockIndexes.push(lineIndex)
+        lineIndex++
+      }
+      if (line.text === '[sidebar]') {
+        openBlockIndexes.push(lineIndex, lineIndex + 2)
```

Comment:
I guess you are using `+ 2` to offset the block title? Any open block can have a title so something must be wrong since we don't do it for `[source]` 🤔